library/src/main/java/org/apache/apex/malhar/lib/state/spillable/SpillableMapImpl.java

```diff
@@ -236,13 +237,27 @@ public void beginWindow(long windowId)
   @Override
   public void endWindow()
   {
+    boolean isTimeUnifiedStore = (store instanceof ManagedTimeUnifiedStateImpl);
```

Comment:
Why do we need to recompute it in every end window? I think you can make it a class variable. It can be transient and initialized in the setup method.